task(upgrade): write upgrade guide and bash script (#437)
bugsnag/bugsnag-unity: UPGRADING.md

#### Deprecations

The following property has been removed from the `Bugsnag` client:

| v5.x API                                                           | v6.x API                                                          |
| ------------------------------------------------------------------ | ----------------------------------------------------------------- |
| `Bugsnag.Metadata`                                                 | Deprecated - no longer public API.                                |

### Breadcrumbs

#### Deprecations

The following properties/methods have been removed from the `Bugsnag` client:

| v5.x API                                                           | v6.x API                                                          |
| ------------------------------------------------------------------ | ----------------------------------------------------------------- |
| `Bugsnag.LeaveBreadcrumb(message)`                                 | Replaced by `Bugsnag.LeaveBreadcrumb(message, metadata, type)`    |
| `Bugsnag.LeaveBreadcrumb(message, type, metadata)`                 | Replaced by `Bugsnag.LeaveBreadcrumb(message, metadata, type)`    |

> | `Bugsnag.LeaveBreadcrumb(breadcrumb)`                              | Replaced by `Bugsnag.LeaveBreadcrumb(message, metadata, type)`    |

Reviewer: The second and third arguments are optional, so I don't think it's true to say that `LeaveBreadcrumb(message)` is deprecated?

Author: Fixed 👍

### Callbacks

#### Additions

The following methods have been added to the `Bugsnag` class: 

| Property/Method                                                    | Usage                                                             |
| ------------------------------------------------------------------ | ----------------------------------------------------------------- |

> | `AddOnError` | Add a callback to modify or discard a event before it is recorded.  LINK_DOCS   | 

Reviewer: ```suggestion
| `AddOnError` | Add a callback to modify or discard an event before it is captured (Android JVM exceptions, ANRs and handled events only).  LINK_DOCS   | 
```

> |  `AddOnSendError`  | Add a callback to modify or discard a event before it is sent (including native Cocoa and Android events).  LINK_DOCS   | 

Reviewer: ```suggestion
| `AddOnSendError`  | Add a callback to modify or discard any event before it is sent (including all iOS/macOS and native Android crashes).  LINK_DOCS   | 
```

| v5.x API                                                           | v6.x API                                                          |
| ------------------------------------------------------------------ | ----------------------------------------------------------------- |
| `Bugsnag.BeforeNotify`                                             | Replaced by `Bugsnag.AddOnSendError(callback) or Bugsnag.AddOnError(callback)`|

### Misc

#### Deprecations

The following methods have been removed from the `Bugsnag` client:

| v5.x API                                                           | v6.x API                                                          |
| ------------------------------------------------------------------ | ----------------------------------------------------------------- |
| `Bugsnag.SetAutoNotify(bool)`                                      | Deprecated - no longer public API.  Callbacks should now be used to stop sending particular errors at runtime|
| `Bugsnag.SetAutoDetectErrors(bool)`                                      | Deprecated - no longer public API.  Callbacks should now be used to stop sending particular errors at runtime|

> | `Bugsnag.SetAutoDetectAnrs(bool)`                                  | Deprecated - no longer public API.  Callbacks should now be used to stop sending particular errors at runtime|

Reviewer: We should say these are replaced by the `AutoDetectErrors` and `EnabledErrorTypes` configuration options or controlled in an on-send callback.

Author: Done 👍
